src/components/fields/date-range-field/date-range-field.tsx

```diff
@@ -75,6 +78,23 @@ export const DateRangeField = (props: IGenericFieldProps<TDateRangeFieldSchema>)
 						!!DateTimeHelper.toLocalDateOrTime(value.to, dateFormat, "date")
 					);
 				})
+				.test(
+					"number-of-days",
+					noOfDaysRule?.["errorMessage"] ||
+						ERROR_MESSAGES.DATE_RANGE.MUST_BE_WITHIN_NUMBER_OF_DAYS(noOfDaysRule?.["numberOfDays"]),
+					(value) => {
+						if (variant === "week") return true;
+						if (!isValidDate(value.from) || !isValidDate(value.to) || !noOfDaysRule?.["numberOfDays"])
+							return true;
+						const localDateFrom = DateTimeHelper.toLocalDateOrTime(value.from, dateFormat, "date");
+						const localDateTo = DateTimeHelper.toLocalDateOrTime(value.to, dateFormat, "date");
+						try {
+							return localDateTo.equals(localDateFrom.plusDays(noOfDaysRule?.["numberOfDays"]));
```

Comment:
should plus `(numberOfDays - 1)` as the start date is inclusive

1 Jan + 7 days = 8 Jan (incorrect)
1 Jan + (7 - 1) days = 7 Jan (expected)